Tic Tac Toe.py

```diff
@@ -0,0 +1,175 @@
+import random
+
+dic = {'1':' ', '2':' ', '3':' ',
+       '4':' ', '5':' ', '6':' ',
+       '7':' ', '8':' ', '9':' '
+       }
+
+def print_template():
+    print('''===========================
+Welcome to Tic Tac Toe
+GAME RULES:
+Each player can place one mark (or stone) per turn on the 3x3 grid
+The WINNER is who succeeds in placing three of their marks in a
+* horizontal,
+* vertical or
+* diagonal row
+Let's start the game''')
+    print('------')
+    print(' | | ')
+    print('------')
+    print(' | | ')
+    print('------')
+    print(' | | ')
+    print('------')
+    print('='*27)
+
+def first_player():
+       lst = [0, 1]
+       random.shuffle(lst)
+       return lst
+
+
+def choose_sign(player1, player2):
+    while True:
+        if player1 == 1:
+            player = 'Player1'
+            choices = ['x', 'o']
+            sypher = input('Random choose {} as first player. {} please insert your sign: x or o \n'.format(player, player))
+            if sypher != choices[0] and sypher != choices[1]:
+                print('You have to insert x or o')
+                continue
+            else:
+                if sypher == 'x':
+                    return ['x', 'o']
+                else:
+                    return ['o', 'x']
+
+        elif player2 == 1:
+            player = 'Player2'
+            choices = ['x', 'o']
+            sypher = input('Random choose {} as first player. {} please insert your sign: x or o \n'.format(player, player))
+            if sypher != choices[0] and sypher != choices[1]:
+                print('You have to insert x or o ')
+                continue
+            else:
+                if sypher == 'x':
+                    return ['o', 'x']
+                else:
+                    return ['x', 'o']
+
+
+def print_progress(dic):
+    print('------')
+    print('{}|{}|{}'.format(dic['1'], dic['2'], dic['3']))
+    print('------')
+    print('{}|{}|{}'.format(dic['4'], dic['5'], dic['6']))
+    print('------')
+    print('{}|{}|{}'.format(dic['7'], dic['8'], dic['9']))
+    print('------')
+
+
+def win(dic, sypher):
+
+    if dic['1'] == dic['2'] == dic['3'] == sypher:
+        return True
+    if dic['4'] == dic['5'] == dic['6'] == sypher:
+        return True
+    if dic['7'] == dic['8'] == dic['9'] == sypher:
+        return True
+
+    if dic['1'] == dic['4'] == dic['7'] == sypher:
+        return True
+    if dic['2'] == dic['5'] == dic['8'] == sypher:
+        return True
+    if dic['3'] == dic['6'] == dic['9'] == sypher:
+        return True
+
+    if dic['1'] == dic['5'] == dic['9'] == sypher:
+        return True
+    if dic['3'] == dic['5'] == dic['7'] == sypher:
+        return True
```

Comment:
You can "compress" the information about winning configurations to a list of lists:
```python
WINNING_CONFIGURATIONS = [
  ['1', '2', '3'],
  ['4', '5', '6'],
  ...
  ['3', '5', '7']
]
...
def win(board, sign):
  for x, y, z in WINNING_CONFIGURATION:
    if board[x] == board[y] == board[z] == sign:
      return True
  return False
```
This is just a suggestion but it allows you to have information about winning configurations stored in data not in code.